### PR TITLE
ADA Requests MVP

### PR DESCRIPTION
**Pull request by OxygenCobalt**

### CODE REVIEW

**Review by ezrichards — APPROVED** [review]

File: backend/alembic/versions/38c5326f0334_create_pickup_spots_table.py
Comment (tbwrigh) [inline backend/alembic/versions/38c5326f0334_create_pickup_spots_table.py:1]: I am 99% sure we will want a description for each ADA pick up spot, but we can worry about this later.

File: backend/alembic/versions/8166e12f260c_create_ada_requests_table.py
Comment (tbwrigh) [inline backend/alembic/versions/8166e12f260c_create_ada_requests_table.py:1]: Shouldn't an ADA request have an associated pick up time?

File: backend/src/handlers/ada.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                "longitude": spot.lon,
+            }
+            pickup_spots_json.append(spot_json)
+
+        return pickup_spots_json
+
+
+@router.post("/pickup_spots")
+def post_pickup_spot(spot: PickupSpotModel, req: Request):
+    new_spot = PickupSpot(name=spot.name)
```
Comment (tbwrigh) [inline backend/src/handlers/ada.py:39]: Shouldn't PickupSpot include lat/lon?

File: backend/src/handlers/ada.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+@router.put("/pickup_spots/{id}")
+def update_pickup_spot(id: int, spot: PickupSpotModel, req: Request):
+    with req.app.state.db.session() as session:
+        pickup_spot = session.query(PickupSpot).filter(PickupSpot.id == id).first()
+
+        if pickup_spot is None:
+            raise HTTPException(status_code=404, detail="Pickup spot not found")
+
+        pickup_spot.name = spot.name
```
Comment (tbwrigh) [inline backend/src/handlers/ada.py:56]: Same thing here, shouldn't this be able to change the lat lon of PickupSpot

File: backend/src/handlers/ada.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+from src.request import process_include
+
+router = APIRouter(prefix="/ada", tags=["ada"])
+
+FIELD_PICKUP_SPOTS = "pickup_spot"
+INCLUDES = {FIELD_PICKUP_SPOTS}
+
+
+class PickupSpotModel(BaseModel):
+    name: str
```
Comment (tbwrigh) [inline backend/src/handlers/ada.py:17]: need lat lon in the model too

File: backend/src/handlers/ada.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            )
+            ada_requests = session.query(ADARequest).filter(
+                ADARequest.created_at >= now, ADARequest.created_at <= end_of_day
+            )
+        elif filter == "future":
+            ada_requests = session.query(ADARequest).filter(
+                ADARequest.created_at >= now
+            )
+        else:
+            raise HTTPException(status_code=400, detail=f"Invalid filter {filter}")
```
Comment (tbwrigh) [inline backend/src/handlers/ada.py:107]: Why not have no filter just return all?

File: backend/src/handlers/ada.py [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            .filter(PickupSpot.id == ada_request_model.pickup_spot_id)
+            .count()
+        )
+        if not pickup_spot:
+            raise HTTPException(status_code=400, detail="Pickup spot not found")
+
+        pickup_spot = ADARequest(
+            pickup_spot=ada_request_model.pickup_spot_id,
+            wheelchair=ada_request_model.wheelchair,
+            created_at=pickup_time,
```
Comment (tbwrigh) [inline backend/src/handlers/ada.py]: created at is request time?

File: backend/src/model/ada_request.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+
+class ADARequest(Base):
+    __tablename__ = "ada_requests"
+    __table_args__ = (ForeignKeyConstraint(["pickup_spot"], ["pickup_spots.id"]),)
+    id: Mapped[int] = mapped_column(
+        primary_key=True, autoincrement=True, nullable=False
+    )
+    pickup_spot: Mapped[int] = mapped_column(nullable=False)
+    created_at: Mapped[datetime] = mapped_column(TZDateTime, nullable=False)
```
Comment (tbwrigh) [inline backend/src/model/ada_request.py]: is time for request?

**Review by tbwrigh — APPROVED** [review]

**Pull request merged**